Removing IIS config for each of the web services in our microservices…

avneet723 opened the pull request:
… to remove the IIS config and default to a local web service

# Types of changes
<!-- Please put an `x` (e.g. [x]) in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:
<!-- Please put an `x` (e.g. [x]) in all the boxes that apply: -->
- [x] All new and existing tests passed.
- [x] The code follows the code style and conventions of this project.
- [x] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

# Description of the change
<!-- Please provide enough information so others can review your pull request -->
...

# Motivation for the change
<!-- Please explain the motivation for making this change -->
...

isaac73 commented:
@avneet723 is there a need to change documentation?

isaac73 commented:
This change needs to be done in the below repos as well? 

https://github.com/Azure/pcs-diagnostics-dotnet
https://github.com/Azure/device-simulation-dotnet

avneet723 commented:
> @avneet723 is there a need to change documentation?

No change to the docs.microsoft documentation. We don't explicitly call out IIS anywhere AFAIK, but I'll double check.

dluc commented:
is IIS express causing any problem? as far as I can remember it was useful to VS users

isaac73 commented:
If the user hasn’t installed IIS along with VS, the solution is crashing on Run. We cannot assume that all VS users have installed IIS.

If the developer hasn’t installed, he/she has to manually reset individual microservice settings to ‘webservice’.


  *   Isaac

From: Devis Lucato <notifications@github.com>
Sent: Sunday, October 28, 2018 4:29 PM
To: Azure/remote-monitoring-services-dotnet <remote-monitoring-services-dotnet@noreply.github.com>
Cc: Isaac Swamidasan <Isaac.Joseph@microsoft.com>; Review requested <review_requested@noreply.github.com>
Subject: Re: [Azure/remote-monitoring-services-dotnet] Removing IIS config for each of the web services in our microservices… (#173)


is IIS express causing any problem? as far as I can remember it was useful to VS users

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fremote-monitoring-services-dotnet%2Fpull%2F173%23issuecomment-433752580&data=02%7C01%7CIsaac.Joseph%40microsoft.com%7Cad06ea2731014882307908d63d2d1adf%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636763661268973132&sdata=MxpIRhZg%2Bl1LKhr0lkULR%2FL6l1xl%2B3zbKnpU5dBtKLE%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAU_njYu_I5_HrybPaif_82qyBWRCEnu7ks5upj2sgaJpZM4X7IOe&data=02%7C01%7CIsaac.Joseph%40microsoft.com%7Cad06ea2731014882307908d63d2d1adf%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636763661268983136&sdata=2mB%2F3B7JwfsCFhn65329y3YiRS%2FwuK2sEI9vszrolso%3D&reserved=0>.

isaac73 commented:
> is IIS express causing any problem? as far as I can remember it was useful to VS users

If the user hasn’t installed IIS along with VS, the solution is crashing on Run. We cannot assume that all VS users have installed IIS.

If the developer hasn’t installed, he/she has to manually reset individual microservice settings to ‘webservice’.

avneet723 commented:
> is IIS express causing any problem? as far as I can remember it was useful to VS users

Two issues: 
- Services running IIS are unable to communicate with service (Device Simulation) running under Docker. 
- There are IIS config issues which fail the service startup. 

From talking to @ppathan, it didn't look like we ever supported IIS.

ppathan commented:
To be clear we did support IIS, but the recent problems that have surfaced as mentioned above it makes sense to remove it since it will be selected as default option when available.

avneet723 commented:
Looks like launchSettings.json for Auth service was in .gitignore file so the change for that WebService didn't go through. Just updated the PR to add the file to source and updated it.

avneet723 commented:
Thanks for the review folks. Closing.
